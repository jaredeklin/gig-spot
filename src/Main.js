import React from 'react';
import { TonightCard } from './TonightCard';
import { connect } from 'react-redux';

export const Main = (props) => {
  let tonightCards;

  if (props.shows) {
    tonightCards = props.shows.map(show => {
      return(<TonightCard show={show} key={show.id} />)
    });
  };

  return (
    <div className='main'>
      <section className='tonights-shows'>
        <div className='tonight-inner'>
          { tonightCards }
        </div>
      </section>
    </div>
  );
};

export const mapStateToProps = (state) => ({
  shows: state.shows
});

export default connect(mapStateToProps)(Main);