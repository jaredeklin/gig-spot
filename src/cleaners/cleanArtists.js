export class CleanArtists {
  clean = (show) => {

    if (!show.performers) {
      const name = show.title.replace(/ [(][0-9][0-9][+] Event[)]/gi, "");
      
      return [name];
    }

    if (show.performers.performer.name) {
      return [show.performers.performer.name];
    }

    return show.performers.performer.map(artist => artist.name);
  };
}