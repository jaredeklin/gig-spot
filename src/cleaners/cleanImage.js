export const cleanImage = (artistData) => {
  if (!artistData.artist || !artistData.artist.image[4]) {

    return null; 
  } else {
    const artistImage = artistData.artist.image[4][`#text`];

    return artistImage;
  }
};