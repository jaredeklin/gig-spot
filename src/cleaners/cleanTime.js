export const cleanTime = (time) => {
  // console.log(time)
  const startTime = new Date(time).toLocaleTimeString([], {
    hour: '2-digit', 
    minute: '2-digit'
  });
  // console.log(startTime)
  return (startTime === '12:00 AM' ? '' : startTime);
};