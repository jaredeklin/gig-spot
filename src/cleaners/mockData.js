export const mockFetchShowsData = {
    "Info": {
        "TotalResults": 174,
        "PageNumber": 0,
        "Message": null
    },
    "Events": [{
        "Id": 3106726,
        "Date": "2018-04-08T21:00:00",
        "Venue": {
            "Id": 160831,
            "Name": "Streets Of London Pub",
            "Address": "1501 E Colfax Ave",
            "City": "Denver",
            "State": "Colorado",
            "StateCode": "CO",
            "Country": "US",
            "CountryCode": "US",
            "ZipCode": "80218",
            "Url": "",
            "Latitude": 0.0,
            "Longitude": 0.0
        },
        "Artists": [{
            "Id": 123882,
            "Name": "Shawn James"
        }],
        "TicketUrl": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1634017/tfly?utm_medium=api"
    }]
}

export const mockCleanConcertData = [{
        "Id": 3106726,
        "Date": "2018-04-08T21:00:00",
        "Venue": {
            "Id": 160831,
            "Name": "Streets Of London Pub",
            "Address": "1501 E Colfax Ave",
            "City": "Denver",
            "State": "Colorado",
            "StateCode": "CO",
            "Country": "US",
            "CountryCode": "US",
            "ZipCode": "80218",
            "Url": "",
            "Latitude": 0.0,
            "Longitude": 0.0
        },
        "Artists": [{
            "Id": 123882,
            "Name": "Shawn James"
        }],
        "TicketUrl": "http://www.shareasale.com/r.cfm?u=460319&b=234786&m=27601&afftrack=&urllink=https://www.ticketfly.com/purchase/event/1634017/tfly?utm_medium=api"
    }]

export const mockTonightCardData = {
  artist: "Selina Albright", 
  venue: "The Soiled Dove Underground", 
  date: {dayName: "Fri", day: "20", month: "Apr", year: "2018", startTime: "20:00"}, 
  id: 3108932, 
  image: "https://lastfm-img2.akamaized.net/i/u/300x300/2254265bbd13a9aea946244144459709.png" 
};

export const mockTonightCardDataWOImage = {
  artist: "Selina Albright", 
  venue: "The Soiled Dove Underground", 
  date: {dayName: "Fri", day: "20", month: "Apr", year: "2018", startTime: "20:00"}, 
  id: 3108932, 
  image: null
};

export const mockFetchImageConcertData = [{
  artist: "Shawn James", venue: "Streets Of London Pub", date: "Apr 8", startTime: "9:00 PM", id: 3106726
}];

export const mockFetchArtistImageReturnData = {
  artist: {
    bio: {links: {}, published: "18 Sep 2012, 01:00", summary: "Shawn James" },
    image: [
      {'#text': "https://lastfm-img2.akamaized.net/i/u/34s/ef281d679fd4707227ec33aaa633ca95.png", size: "small"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/64s/ef281d679fd4707227ec33aaa633ca95.png", size: "medium"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/174s/ef281d679fd4707227ec33aaa633ca95.png", size: "large"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: "extralarge"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: "mega"},
      {'#text': "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", size: ""}
      ],
    name: "Shawn James",
  }
}

export const mockFetchImageReturnData = [{"artist": "Shawn James", "date": "Apr 8", "id": 3106726, "image": "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png", "startTime": "9:00 PM", "venue": "Streets Of London Pub"}]