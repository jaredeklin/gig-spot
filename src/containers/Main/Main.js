import React from 'react';
import { TonightCard } from '../../components/TonightCard/TonightCard';
import { connect } from 'react-redux';
import LocationForm from '../LocationForm/LocationForm';
import PropTypes from 'prop-types';

export const Main = (props) => {
  const { tonightsShows, thisWeeksShows, upcomingShows, loading } = props;
  const combined = [...tonightsShows, ...thisWeeksShows];
  let tonightCards, thisWeekCards, upcomingCards;
 
  if (tonightsShows) {
    tonightCards = tonightsShows.map(show => {
      return (<TonightCard show={show} key={show.id} />);
    });
  }

  if (thisWeeksShows) {
    if (tonightsShows.length <= 2) {
      thisWeekCards = combined.map(show => {
        return (<TonightCard show={show} key={show.id} />);
      });
    } else {
      thisWeekCards = thisWeeksShows.map(show => {
        return (<TonightCard show={show} key={show.id} />);
      });
    }
  }

  if (upcomingShows) {
    upcomingCards = upcomingShows.map(show => {
      return (<TonightCard show={show} key={show.id} />);
    });
  }

  return (
      
    <div className='main'>
      { !loading && 
          <div className='change-location'>
            <p>Update location:</p>
            <LocationForm id='main-form' />
          </div>
      }
      {
        thisWeeksShows.length > 0 &&
        <div className='all-shows'>
          { tonightsShows.length > 2 &&
          <div className='tonight-outer'>
            <h2 className='event-happening-when-text'>Tonight:</h2>
            <section className='tonights-shows'>
              <div className='shows-inner'>
                { tonightCards }
              </div>
            </section>
          </div>
          }
          <h2 className='event-happening-when-text'>This Week:</h2>
          <section className='shows'>
            <div className='shows-inner'>
              { thisWeekCards }
            </div>
          </section>
          { 
            upcomingShows.length > 3 &&
              <div>
                <h2 className='event-happening-when-text'>Upcoming:</h2>
                <section className='shows'>
                  <div className='shows-inner'>
                    { upcomingCards }
                  </div>
                </section>
              </div>
          }
          <div className="jambase">
            <a 
              href="http://www.JamBase.com" 
              target="_top" 
              title="JamBase Concert Search"
            ><img 
                src= "http://images.jambase.com/logos/jambase140x70.gif" 
                alt="Search JamBase Concerts" 
                border="0" />
            </a>
          </div>
        </div>
      }
    </div>
  );
};

export const mapStateToProps = (state) => {
  const { tonightsShows, thisWeeksShows, upcomingShows, loading } = state;
  
  return {
    tonightsShows,
    thisWeeksShows,
    upcomingShows,
    loading
  };
};

export default connect(mapStateToProps)(Main);

Main.propTypes = {
  tonightsShows: PropTypes.array,
  thisWeeksShows: PropTypes.array,
  upcomingShows: PropTypes.array,
  loading: PropTypes.bool
};