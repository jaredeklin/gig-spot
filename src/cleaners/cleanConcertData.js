import { CleanArtists } from './cleanArtists';
import { Dates } from './Dates';
const moment = require('moment');

const cleanArtists = new CleanArtists();
const time = new Dates();

export const cleanConcertData = concerts => {
  concerts = concerts.filter(concert => {
    const { title, performers } = concert;

    if (title.includes('Tickets') && !performers) {
      return false;
    }

    if (title.includes('Parking')) {
      return false;
    }
    return true;
  });

  return concerts.reduce((concertArray, show) => {
    const {
      id,
      popularity,
      tickets,
      description,
      image,
      start_time,
      categories,
      links
    } = show;

    const venue = {
      name: show.venue_name,
      id: show.venue_id,
      url: show.venue_url,
      address: show.venue_address,
      city: show.city_name,
      state: show.region_abbr,
      zip: show.postal_code
    };
    const artists = cleanArtists.clean(show);
    const headlineArtist = artists[0];
    const supportArtists = artists.filter(artist => artist !== artists[0]);
    const ticketUrl = cleanTickets(links, tickets);

    const date = moment(start_time).format('MMM D');
    const startTime = time.cleanTime(start_time);
    const concertData = {
      headlineArtist,
      supportArtists,
      description,
      venue,
      date,
      startTime,
      id,
      popularity,
      tickets: ticketUrl,
      image: !image ? null : image.block250.url,
      categories
    };

    return [...concertArray, concertData];
  }, []);
};

const cleanTickets = (links, tickets) => {
  if (links) {
    const preferred = links.link.find(ticket => ticket.url.includes('axs.com'));

    if (preferred) {
      return preferred.url;
    }
  }

  if (tickets) {
    return tickets.link[0].url;
  }

  return null;
};
