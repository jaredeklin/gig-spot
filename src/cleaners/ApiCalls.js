import { SimpleCleaners } from './SimpleCleaners';
import { cleanConcertData } from './cleanConcertData';

const lastFmApiKey = process.env.REACT_APP_LASTFM_KEY;
const clean = new SimpleCleaners();

export class ApiCalls {
  getLastFmData = concerts => {
    const promises = concerts.map(async concert => {
      const artist = concert.headlineArtist.replace(/&/gi, 'and');
      const baseUrl = 'http://ws.audioscrobbler.com/2.0/?method=artist.getinfo';
      const keyUrl = `&artist=${artist}&api_key=${lastFmApiKey}&format=json`;

      try {
        const response = await fetch(`${baseUrl}${keyUrl}`);
        const artistData = await response.json();
        const image = clean.image(artistData, concert.image);
        const bio = clean.bio(artistData);

        return { ...concert, image, bio };
      } catch (Error) {
        console.log(Error); //eslint-disable-line
        return { ...concert, bio: null };
      }
    });

    return Promise.all(promises);
  };

  getEvents = async (type, url, date) => {
    const dateRange = this.getDateRange(type, date);
    const response = await fetch(`${url}&date=${dateRange}`);

    if (!response.ok) {
      throw Error(response.statusText);
    }

    return await this.cleanData(response);
  };

  cleanData = async response => {
    const concertData = await response.json();
    const cleanData = cleanConcertData(concertData.events.event);

    return await this.getLastFmData(cleanData);
  };

  getDateRange = (type, date) => {
    const { tommorrow, nextWeek, upcoming, upcomingEnd } = date;
    switch (type) {
      case 'today':
        return 'Today';

      case 'week':
        return `${tommorrow}-${nextWeek}`;

      case 'upcoming':
        return `${upcoming}-${upcomingEnd}`;

      default:
        console.log('Invalid type param'); //eslint-disable-line
        break;
    }
  };
}
