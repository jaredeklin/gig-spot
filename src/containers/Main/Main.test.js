import React from 'react';
import { shallow } from 'enzyme';
import { Main, mapStateToProps } from './Main';
import { mockFetchImageReturnData } from '../../cleaners/mockData';

describe('Main', () => {
  const mockShows = { shows: mockFetchImageReturnData}

  it('should match the snapshot with props', () => {
    let wrapper = shallow(<Main props={mockShows}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    let wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should map state to props', () => {
    let wrapper = shallow(<Main />);
    const mockShows = { shows: { artist: 'someone' }};

    const mapped = mapStateToProps(mockShows);
    expect(mapped.shows).toEqual(mockShows.shows);
  });
});