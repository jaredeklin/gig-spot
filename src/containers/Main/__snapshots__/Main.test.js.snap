// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="tonights-shows"
>
        <div
                className="tonight-inner"
        >
                <h4>
                        Tonight:
                </h4>
        </div>
</section>,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tonight-inner"
>
          <h4>
                    Tonight:
          </h4>
</div>,
        "className": "tonights-shows",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Tonight:
</h4>,
            undefined,
          ],
          "className": "tonight-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tonight:",
            },
            "ref": null,
            "rendered": "Tonight:",
            "type": "h4",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="tonights-shows"
>
          <div
                    className="tonight-inner"
          >
                    <h4>
                              Tonight:
                    </h4>
          </div>
</section>,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="tonight-inner"
>
            <h4>
                        Tonight:
            </h4>
</div>,
          "className": "tonights-shows",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Tonight:
</h4>,
              undefined,
            ],
            "className": "tonight-inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tonight:",
              },
              "ref": null,
              "rendered": "Tonight:",
              "type": "h4",
            },
            undefined,
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main should match the snapshot with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    props={
        Object {
            "shows": Array [
              Object {
                "artist": "Shawn James",
                "date": "Apr 8",
                "id": 3106726,
                "image": "https://lastfm-img2.akamaized.net/i/u/300x300/ef281d679fd4707227ec33aaa633ca95.png",
                "startTime": "9:00 PM",
                "venue": "Streets Of London Pub",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="tonights-shows"
>
        <div
                className="tonight-inner"
        >
                <h4>
                        Tonight:
                </h4>
        </div>
</section>,
      "className": "main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="tonight-inner"
>
          <h4>
                    Tonight:
          </h4>
</div>,
        "className": "tonights-shows",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Tonight:
</h4>,
            undefined,
          ],
          "className": "tonight-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tonight:",
            },
            "ref": null,
            "rendered": "Tonight:",
            "type": "h4",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="tonights-shows"
>
          <div
                    className="tonight-inner"
          >
                    <h4>
                              Tonight:
                    </h4>
          </div>
</section>,
        "className": "main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="tonight-inner"
>
            <h4>
                        Tonight:
            </h4>
</div>,
          "className": "tonights-shows",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Tonight:
</h4>,
              undefined,
            ],
            "className": "tonight-inner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tonight:",
              },
              "ref": null,
              "rendered": "Tonight:",
              "type": "h4",
            },
            undefined,
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
